src/test/kfs_rest_parser_test.cpp

```diff
@@ -0,0 +1,769 @@
+
+    auto parameter = proto.outputs()[0].parameters().find("param");
+    ASSERT_NE(parameter, proto.outputs()[0].parameters().end());
+    ASSERT_EQ(parameter->second.parameter_choice_case(), inference::InferParameter::ParameterChoiceCase::kBoolParam);
+    ASSERT_EQ(parameter->second.bool_param(), true);
+}
+
+TEST_F(KFSRestParserTest, parseValidRequestWithStringInputParameter) {
+    std::string request = R"({
+    "inputs" : [
+        {
+        "name" : "input0",
+        "shape" : [ 2, 2 ],
+        "datatype" : "UINT32",
+        "data" : [ 1, 2, 3, 4 ],
+        "parameters" : {"param" : "value"}
+        }
+    ]
+    })";
+    auto status = parser.parse(request.c_str());
+    ASSERT_EQ(status, StatusCode::OK);
+
+    VALIDATE_INPUT("UINT32", uint_contents_size, uint_contents);
+
+    auto parameter = proto.inputs()[0].parameters().find("param");
+    ASSERT_NE(parameter, proto.inputs()[0].parameters().end());
+    ASSERT_EQ(parameter->second.parameter_choice_case(), inference::InferParameter::ParameterChoiceCase::kStringParam);
+    ASSERT_EQ(parameter->second.string_param(), "value");
+}
+
+TEST_F(KFSRestParserTest, parseValidRequestWithIntInputParameter) {
+    std::string request = R"({
+    "inputs" : [
+        {
+        "name" : "input0",
+        "shape" : [ 2, 2 ],
+        "datatype" : "UINT32",
+        "data" : [ 1, 2, 3, 4 ],
+        "parameters" : {"param" : 5}
+        }
+    ]
+    })";
+    auto status = parser.parse(request.c_str());
+    ASSERT_EQ(status, StatusCode::OK);
+
+    VALIDATE_INPUT("UINT32", uint_contents_size, uint_contents);
+
+    auto parameter = proto.inputs()[0].parameters().find("param");
+    ASSERT_NE(parameter, proto.inputs()[0].parameters().end());
+    ASSERT_EQ(parameter->second.parameter_choice_case(), inference::InferParameter::ParameterChoiceCase::kInt64Param);
+    ASSERT_EQ(parameter->second.int64_param(), 5);
+}
+
+TEST_F(KFSRestParserTest, parseValidRequestWithBoolInputParameter) {
+    std::string request = R"({
+    "inputs" : [
+        {
+        "name" : "input0",
+        "shape" : [ 2, 2 ],
+        "datatype" : "UINT32",
+        "data" : [ 1, 2, 3, 4 ],
+        "parameters" : {"param" : true}
+        }
+    ]
+    })";
+    auto status = parser.parse(request.c_str());
+    ASSERT_EQ(status, StatusCode::OK);
+
+    VALIDATE_INPUT("UINT32", uint_contents_size, uint_contents);
+
+    auto parameter = proto.inputs()[0].parameters().find("param");
+    ASSERT_NE(parameter, proto.inputs()[0].parameters().end());
+    ASSERT_EQ(parameter->second.parameter_choice_case(), inference::InferParameter::ParameterChoiceCase::kBoolParam);
+    ASSERT_EQ(parameter->second.bool_param(), true);
+}
+
+TEST_F(KFSRestParserTest, parseValidRequestWithNoDataButBinaryInputsParameter) {
+    std::string request = R"({
+    "inputs" : [
+        {
+        "name" : "input0",
+        "shape" : [ 2, 2 ],
+        "datatype" : "BYTES",
+        "parameters" : {"binary_data_size" : 16}
```

Comment:
it's irrelevant from the parser perspective, but it looks like this size does not match shape and datatype